app/src/main/java/fr/gaulupeau/apps/Poche/network/WallabagService.java

```diff
@@ -33,51 +36,60 @@ public WallabagService(String endpoint, String username, String password) {
         this(endpoint, username, password, WallabagConnection.getClient(), -1);
     }
 
-    public WallabagService(String endpoint, String username, String password, OkHttpClient client, int wallabagMajorVersion) {
+    public WallabagService(String endpoint, String username, String password, OkHttpClient client,
+                           int wallabagMajorVersion) {
         this.endpoint = endpoint;
         this.client = client;
+        this.username = username;
+        this.password = password;
         this.wallabagMajorVersion = wallabagMajorVersion;
         this.settings = App.getInstance().getSettings();
-
-        if(this.wallabagMajorVersion == -1) {
-            this.wallabagMajorVersion = guessWallabagVersion();
-        }
-
-        switch (this.wallabagMajorVersion) {
-            case 1:
-                serviceEndpoint = new WallabagServiceEndpointV1(endpoint, username, password, client);
-                break;
-            case 2:
-                serviceEndpoint = new WallabagServiceEndpointV2(endpoint, username, password, client);
-                break;
-            default:
-                Log.w(TAG, "switch default case for wallabagMajorVersion reached. this.wallabagMajorVersion=" + this.wallabagMajorVersion);
-                serviceEndpoint = new WallabagServiceEndpointV1(endpoint, username, password, client);
-        }
     }
 
     public FeedsCredentials getCredentials() throws IOException {
-        return serviceEndpoint.getCredentials();
+        return getServiceEndpoint().getCredentials();
     }
 
     public boolean addLink(String link) throws IOException {
-        return serviceEndpoint.addLink(link);
+        return getServiceEndpoint().addLink(link);
     }
 
     public boolean toggleArchive(int articleId) throws IOException {
-        return serviceEndpoint.toggleArchive(articleId);
+        return getServiceEndpoint().toggleArchive(articleId);
     }
 
     public boolean toggleFavorite(int articleId) throws IOException {
-        return serviceEndpoint.toggleFavorite(articleId);
+        return getServiceEndpoint().toggleFavorite(articleId);
     }
 
     public boolean deleteArticle(int articleId) throws IOException {
-        return serviceEndpoint.deleteArticle(articleId);
+        return getServiceEndpoint().deleteArticle(articleId);
     }
 
     public int testConnection() throws IOException {
-        return serviceEndpoint.testConnection();
+        return getServiceEndpoint().testConnection();
+    }
+
+    private WallabagServiceEndpoint getServiceEndpoint() {
+        if(serviceEndpoint == null) {
+            if(this.wallabagMajorVersion == -1) {
+                this.wallabagMajorVersion = guessWallabagVersion();
+            }
+
+            switch (this.wallabagMajorVersion) {
+                case 2:
```

Comment:
@nicosomb it's not a problem cause there is no break; in the default case. Thus it falls through to case 1. So if wallabagMajorVersion is anything but one or two, it also uses V1 endpoint.

@di72nn Shouldn't we make V2 the default, because it is the current release?